xline/src/storage/kvstore.rs

```diff
@@ -38,16 +39,46 @@ struct KvStoreInner {
     btree: Mutex<BTreeMap<Vec<u8>, KeyValue>>,
 }
 
+/// Execution Request
+#[derive(Debug)]
+struct ExecutionRequest {
+    /// Command to execute
+    cmd: Command,
+    /// Command request sender
+    res_sender: oneshot::Sender<CommandResponse>,
+}
+
+impl ExecutionRequest {
+    /// New `ExectionRequest`
+    fn new(cmd: Command) -> (Self, oneshot::Receiver<CommandResponse>) {
+        let (tx, rx) = oneshot::channel();
+        (
+            Self {
+                cmd,
+                res_sender: tx,
+            },
+            rx,
+        )
+    }
+
+    /// Consume `ExecutionRequest` and get ownership of each field
+    fn unpack(self) -> (Command, oneshot::Sender<CommandResponse>) {
+        let Self { cmd, res_sender } = self;
+        (cmd, res_sender)
+    }
+}
+
 impl KvStore {
     /// New `KvStore`
     pub(crate) fn new() -> Self {
-        let (request_tx, mut request_rx) = channel(CHANNEL_SIZE);
+        let (request_tx, mut request_rx) = mpsc::channel(CHANNEL_SIZE);
         let inner = Arc::new(KvStoreInner::new());
         let inner_clone = Arc::clone(&inner);
         let _handle = tokio::spawn(async move {
             let mut btree = inner.btree.lock().await;
             while let Some(req) = request_rx.recv().await {
```

Comment:
Can we put the btree into the while loop so that we can protect btree with sync lock, which is much faster?

Reply:
So far this is temp solution, because this is the only position to modify btree, so I get lock first and then pass guard. Will change to get lock inside the loop.